app.py

```diff
@@ -1,53 +1,146 @@
 import streamlit as st
 import pandas as pd
-import time
-
-# def loader():
-#     latest_iteration = st.empty()
-#     bar = st.progress(0)
+import json
+import data_upload_DHIS2 as dhis2
 
 # Initializing variables
 form_types = ("ICCM", "OPD", "RHGynobs", "Vaccination")
+# Ideally, these could be set in this app?
+# dhis2_username = ''
+# dhis2_password = ''
+# DHIS2_Test_Server_URL = ''
 
-st.title("MSF OCR Tool")
 
-st.write("### File Upload ###")
+# Function definitions!
+# """ Creates a list of dataValues by going through each cell
+# and recording the column, header, and cell value of non-None
 
-# results = ocr.run()
+# :param df: an iterable of data frames
+# :return: a list of JSON-ready dataValues
+# """
+def data_values(dfs):
+    data_values = []
+    for df in dfs:
+        columns = df.columns.tolist()
+        for col, i in df.iterrows():
+            for cell in columns:
+                if i[cell] is not None:
+                    new_cell = {"dataElement": f"{col}/{cell}", "value": f"{i[cell]}"}
+                    data_values.append(new_cell)
+    return data_values
+            
+# @st.cache_data
+# Ideally this would cache data but it messes with filling in info
+# """ Creates a JSON payload through the form inputs and, if not empty,
+# calling the getUID command to find the proper ID.
+
+# :param df: an iterable of data frames
+# :return: the JSON payload
+# """

```

Comment:
Thanks for adding docstrings! For python, docstring comments always go after the function/class signature. In this case, pleas just copy the docstring to the line after `def convert(dfs)`. You can see more info at https://realpython.com/documenting-python-code/ and https://peps.python.org/pep-0257/